src/app/modules/submission/receiver/receiver.middleware.ts

```diff
@@ -0,0 +1,49 @@
+import { FieldResponse, FormResponseMode } from '../../../../../shared/types'
+import { createLoggerWithLabel } from '../../../config/logger'
+import { createReqMeta } from '../../../utils/request'
+import { ControllerHandler } from '../../core/core.types'
+
+import * as SubmissionReceiver from './receiver.service'
+import { mapRouteError } from './receiver.utils'
+
+const logger = createLoggerWithLabel(module)
+
+/**
+ * Parses multipart-form data request. Parsed attachments are
+ * placed into req.attachments and parsed fields are placed into
+ * req.body.
+ *
+ * @param req - Express request object
+ * @param res - Express response object
+ * @param next - Express next middleware function
+ */
+export const receiveSubmission: ControllerHandler<
+  unknown,
+  { message: string },
+  { responses: FieldResponse[] }
+> = async (req, res, next) => {
+  const logMeta = {
+    action: 'receiveSubmission',
+    ...createReqMeta(req),
+  }
+  const emailMode = req.route.path.endsWith(`/${FormResponseMode.Email}`)
```

Comment:
For this, could I suggest instead of the middleware matching the route, we can use a higher order function

```
makeReceiveSubmissionMiddleware = (isEmailMode) => 
// returns the receiveSubmissionMiddleware with the correct filecreateMultipartReceiver
```

which returns the receiver. Then the controller passes in the correct mode when calling the higher order function

Reply:
Okay I've refactored this in [f167b9f](https://github.com/opengovsg/FormSG/pull/6587/commits/f167b9f0c36814ea8126c7a6c1a4ba50f5f2a090). Let me know if this is fine!